docs/Containerfile.5.md

```diff
@@ -0,0 +1,511 @@
+  image. Consider a `podman build` without the --build-arg flag:
+
+  ```
+  $ podman build Containerfile
+  ```
+
+  Using this Containerfile example, `CONT_IMG_VER` is still persisted in the image but
+  its value would be `v1.0.0` as it is the default set in line 3 by the `ENV` instruction.
+
+  The variable expansion technique in this example allows you to pass arguments
+  from the command line and persist them in the final image by leveraging the
+  `ENV` instruction. Variable expansion is only supported for [a limited set of
+  Containerfile instructions.](#environment-replacement)
+
+  Container engines have a set of predefined `ARG` variables that you can use without a
+  corresponding `ARG` instruction in the Containerfile.
+
+  * `HTTP_PROXY`
+  * `http_proxy`
+  * `HTTPS_PROXY`
+  * `https_proxy`
+  * `FTP_PROXY`
+  * `ftp_proxy`
+  * `NO_PROXY`
+  * `no_proxy`
+  * `ALL_PROXY`
+  * `all_proxy`
+
+  To use these, pass them on the command line using `--build-arg` flag, for
+  example:
+
+  ```
+  $ podman build --build-arg HTTPS_PROXY=https://my-proxy.example.com .
+  ```
+
+**ONBUILD**
+  -- `ONBUILD [INSTRUCTION]`
+  The **ONBUILD** instruction adds a trigger instruction to an image. The
+  trigger is executed at a later time, when the image is used as the base for
+  another build. Container engines execute the trigger in the context of the downstream
+  build, as if the trigger existed immediately after the **FROM** instruction in
+  the downstream Containerfile.
+
+  You can register any build instruction as a trigger. A trigger is useful if
+  you are defining an image to use as a base for building other images. For
+  example, if you are defining an application build environment or a daemon that
+  is customized with a user-specific configuration.
+
+  Consider an image intended as a reusable python application builder. It must
+  add application source code to a particular directory, and might need a build
+  script called after that. You can't just call **ADD** and **RUN** now, because
+  you don't yet have access to the application source code, and it is different
+  for each application build.
+
+  -- Providing application developers with a boilerplate Containerfile to copy-paste
+  into their application is inefficient, error-prone, and
+  difficult to update because it mixes with application-specific code.
+  The solution is to use **ONBUILD** to register instructions in advance, to
+  run later, during the next build stage.
+
+## SEE ALSO
+buildah(1), podman(1), docker(1)
+
+# HISTORY
+```
+May 2014, Compiled by Zac Dover (zdover at redhat dot com) based on docker.com Dockerfile documentation.
+Feb 2015, updated by Brian Goff (cpuguy83@gmail.com) for readability
+Sept 2015, updated by Sally O'Malley (somalley@redhat.com)
+Oct 2016, updated by Addam Hardy (addam.hardy@gmail.com)
+Aug 2021, converted Dockerfile man page to Containerfile by Dan Walsh (dwalsh@redhat.com)
```

Comment:
It might be nice to have a Containers/Containerfiles repo with well documented Containerfiles for commona containers (nginx, onbuild examples, etc.).  Not for the PR, just something we might want to consider.

Reply:
That should be in Buildah if it was there.